### DISCUSSION TITLE
How to get the original message ?

### DISCUSSION THREAD

**Discussion by loveyang2012** [vectordotdev/vector#17831] [Q&A]
The original message like this:
```
{
     "abilityState": "1",
     "t_servicename": "createTask",
     "tag": "push-openapi-access"
}
```
I  get message by vector like this:

```
{
"log": "{\"abilityState\":\"1\",\"t_servicename\":\"createTask\",\"tag\":\"push-openapi-access\"}"
}
```

this is my  transforms  config:

```
[sources.a]
    type = "fluent"
    address = "0.0.0.0:8225"
    receive_buffer_bytes = 65_536
[transforms.b]
type = "remap"
inputs = ["a"]
source = '''
  del(.host)
  del(.source_type)
  del(.timestamp)
  del(.tag)
'''
[sinks.out]
    inputs = ["b"]
    type = "console"
    encoding.codec = "json"
```
How can I get the original message ? 
I used to finish it by  transforms json_parser ,but now it has been deprecated.
```
[transforms.a_parse]
inputs = ["a_coming"]
type = "json_parser"
field = "log"
```

**Comment by jszwedko** [comment]
Hi @loveyang2012 !

You can see examples, for all deprecated and removed transforms, the new way to do it documented here: https://vector.dev/highlights/2022-05-03-0-22-0-upgrade-guide/#deprecated-transforms . In your case, you'd want to use `remap` with the `parse_json` function. Hopefully this helps!
